src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useState } from 'react';
+import { Filter } from '../../types/enumFilter';
+
+type Props = {
+  filterByFlag: (value: Filter) => void;
+  filteredByValue: (value: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterByFlag,
+  filteredByValue,
+}) => {
+  const [showButton, setShowButton] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const handleFilterByFlag = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    filterByFlag(e.target.value as Filter);
+  };
+
+  const handleFilterByValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+    setShowButton(true);
+    filteredByValue(inputValue);
```

Comment:
In `handleFilterByValue`, the `filteredByValue(inputValue)` call uses the state `inputValue` which might not be updated immediately after `setInputValue(e.target.value)`. Consider using `e.target.value` directly in `filteredByValue` to ensure the correct value is passed.